sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/DataSourceV2Strategy.scala

```diff
@@ -120,20 +122,22 @@ class DataSourceV2Strategy(session: SparkSession) extends Strategy with Predicat
         scanExec
```

Comment:
We can't do `if (scanExec.supportsColumnar)` now. We should just call `withProjectAndFilter`

Reply:
Ah, I see, thanks. I've updated.